### PR TITLE
Add types for truffle-hdwallet-provider

### PR DESCRIPTION
**Pull request by LogvinovLeon**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/truffle-hdwallet-provider/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+import {
+    JSONRPCRequestPayload,
+    JSONRPCResponsePayload,
+    Provider
+} from "ethereum-protocol";
+declare class HDWalletProvider implements Provider {
+    constructor(mnemonic: string, rpcUrl: string);
```
Comment (sheetalkamat) [inline types/truffle-hdwallet-provider/index.d.ts]: You want to have third parameter address_index?: number as well as per https://github.com/trufflesuite/truffle-hdwallet-provider#general-usage

File: types/truffle-hdwallet-provider/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    sendAsync(
+        payload: JSONRPCRequestPayload,
+        callback: (err: Error, result: JSONRPCResponsePayload) => void
+    ): void;
+}
```
Comment (sheetalkamat) [inline types/truffle-hdwallet-provider/index.d.ts]: There is no documentation for sendAsync and in https://github.com/trufflesuite/truffle-hdwallet-provider/blob/master/index.js there are more methods than that need to be added if this is added. Cannot verify type from that though so would assume you are typing it correctly.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/truffle-hdwallet-provider/index.d.ts (hunk 1 above)
Comment (LogvinovLeon) [reply types/truffle-hdwallet-provider/index.d.ts]: Added it

File: types/truffle-hdwallet-provider/index.d.ts (hunk 2 above)
Comment (sheetalkamat) [reply types/truffle-hdwallet-provider/index.d.ts]: So are send, getAddress, getAddresses suppose to be private then? Probably add those as private then? I am just looking at the code and seeing if your type reflects it or not.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Comment by miguelmota** [comment]
this type package is needed; hoping something can bring this PR to life again

@LogvinovLeon

**Pull request closed without merging**